tests/test_liquidations_peripheral.py

```diff
@@ -1142,3 +1142,398 @@ def test_admin_withdrawal_wrong_sender(
+    lending_pool_peripheral_contract.setLoansPeripheralAddress(loans_peripheral_contract, {"from": contract_owner})
+    lending_pool_peripheral_contract.setLiquidityControlsAddress(liquidity_controls_contract, {"from": contract_owner})
+
+    erc721_contract.mint(collateral_vault_core_contract, 0, {"from": contract_owner})
+
+    erc20_contract.mint(contract_owner, LOAN_AMOUNT * 2, {"from": contract_owner})
+    erc20_contract.approve(lending_pool_core_contract, LOAN_AMOUNT * 2, {"from": contract_owner})
+    lending_pool_peripheral_contract.deposit(LOAN_AMOUNT * 2, {"from": contract_owner})
+    lending_pool_peripheral_contract.sendFunds(contract_owner, LOAN_AMOUNT, {"from": loans_peripheral_contract})
+
+    erc20_contract.mint(contract_owner, LOAN_AMOUNT * 2, {"from": contract_owner})
+
+    tx_add_loan = loans_core_contract.addLoan(
+        borrower,
+        LOAN_AMOUNT,
+        LOAN_INTEREST,
+        MATURITY,
+        [(erc721_contract, 0, LOAN_AMOUNT)],
+        {"from": loans_peripheral_contract}
+    )
+    loan_id = tx_add_loan.return_value
+    loans_core_contract.updateLoanStarted(borrower, loan_id, {"from": loans_peripheral_contract})
+    loans_core_contract.updateDefaultedLoan(borrower, loan_id, {"from": loans_peripheral_contract})
+
+    liquidations_peripheral_contract.addLiquidation(
+        erc721_contract,
+        0,
+        borrower,
+        loan_id,
+        erc20_contract
+    )
+
+    liquidation = liquidations_peripheral_contract.getLiquidation(erc721_contract, 0)
+    loan = loans_core_contract.getLoan(borrower, loan_id)
+
+    return (liquidation, loan)
+
+
+def test_admin_liquidation(
+    liquidations_peripheral_contract,
+    liquidations_core_contract,
+    loans_peripheral_contract,
+    loans_core_contract,
+    lending_pool_peripheral_contract,
+    lending_pool_core_contract,
+    collateral_vault_peripheral_contract,
+    collateral_vault_core_contract,
+    liquidity_controls_contract,
+    erc721_contract,
+    erc20_contract,
+    borrower,
+    contract_owner
+):
+
+    (liquidation, loan) = _create_liquidation(
+        liquidations_peripheral_contract,
+        liquidations_core_contract,
+        loans_peripheral_contract,
+        loans_core_contract,
+        lending_pool_peripheral_contract,
+        lending_pool_core_contract,
+        collateral_vault_peripheral_contract,
+        collateral_vault_core_contract,
+        liquidity_controls_contract,
+        erc721_contract,
+        erc20_contract,
+        borrower,
+        contract_owner
+    )
+
+    liquidation_id = liquidation['lid']
+    collateral_address = liquidation['collateralAddress']
+    token_id = liquidation['tokenId']
+    loan_id = loan['id']
+
+    erc20_contract.approve(lending_pool_core_contract, LOAN_AMOUNT, {"from": contract_owner})
+    collateral_vault_peripheral_contract.transferCollateralFromLiquidation(contract_owner, collateral_address, token_id, {"from": liquidations_peripheral_contract})
```

Comment:
To create the correct scenario, we should use the `liquidations_peripheral_contract.adminWithdrawal` to get the collateral as the owner and then do the admin liquidation. Does this make sense?

Reply:
The issue is both `adminWithdrawal` and `adminLiquidation` are removing the liquidation. For them to work in tandem, just one should be responsible to do so.
Maybe  `adminWithdrawal` should do it, so we do not have liquidations with no collateral supporting it. What do you think?

Reply:
You're completely right, only `adminWithdrawal` should remove the liquidation. The `adminLiquidation` is always run after the `adminWithdrawal` in the flow.

Reply:
Changed. Now `adminLiquidation` requires two additional parameters (`liquidationId` and `erc20TokenContract`) as it no longer can get them from the liquidation.